calculator_test.go

```diff
@@ -3,27 +3,125 @@ package calculator_test
 import (
 	"calculator"
 	"math"
+	"math/rand"
 	"testing"
+	"time"
 )
 
+func almostEqual(a, b float64) bool {
+	const float64EqualityThreshold = 1e-9
+	return math.Abs(a-b) <= float64EqualityThreshold
+}
+
 func TestAdd(t *testing.T) {
 	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	var tcs = []struct {
```

Comment:
We would probably write `tcs := ...` here, since we are initialising the variable with a non-default value. (If we wanted the default value, then we would write `var x int`, for example.)